src/components/parts/ENSName/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { useState, useEffect } from 'react'
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+
+async function getNameForAddress(address, provider) {
+    const ens = new ENS({ provider, ensAddress: getEnsAddress("1") })
+    return await ens.getName(address)
+  }
+  
+function useENSName(address, provider) {
+    const [ name, setName ] = useState<string>()
+    useEffect(() => {
+        getNameForAddress(address, provider).then(({ name }) => setName(name))
+    })
+    return name
+}
+
+export default function ENSName({ address, fallback = null, provider = window.web3 ? window.web3.currentProvider : window.ethereum }: { address: string, fallback?: string, provider?: any }) {
```

Comment:
I found an error when using Wallet Connect. The provider doesn't get passed into to the `ENSName` component and it triggers a runtime error when it is missing.

One possible solution would be to pass the provider from the `web3Modal` in `ConnectWalletButton.tsx` which would avoid the use of `window` here. I can get this fixed for you if you need help.

![CleanShot 2021-06-16 at 11 55 24](https://user-images.githubusercontent.com/81541283/122261577-2a9ab700-ce9a-11eb-8c41-d5489a7d35f8.png)